File: ANRDemo/app/src/main/java/edu/sfsu/anrdemo/MainActivity.java
package edu.sfsu.anrdemo;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        TextView countdownText = findViewById(R.id.countdown_text);
        Button button = findViewById(R.id.countdown_button);

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                for(int countDown = 10; countDown >= 0; countDown--) {
                    countdownText.setText(countDown + "!");
                    try {
                        Thread.sleep(1000);
                    } catch (InterruptedException e) {
                        e.printStackTrace();
                    }
                }
            }
        });
    }
}

File: DialogFragmentDemo/app/src/main/java/edu/sfsu/dialogfragmentdemo/MainActivity.kt
package edu.sfsu.dialogfragmentdemo

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val button = findViewById<Button>(R.id.tap_me)
        button.setOnClickListener {
            ExampleDialogFragment().show(supportFragmentManager, "tag_name")
        }
    }
}